--- a/pom.xml
+++ b/pom.xml
@@ -151,7 +151,7 @@
         </executions>
         <configuration>
           <version>1.0.0</version>
-          <model>src/main/mdo/antlrOptions.mdo</model>
+          <models><model>src/main/mdo/antlrOptions.mdo</model></models>
         </configuration>
       </plugin>
       <plugin>
@@ -243,4 +243,4 @@
       </build>
     </profile>
   </profiles>
-</project>
\ No newline at end of file
+</project>
--- a/src/main/java/org/codehaus/mojo/antlr/AntlrHtmlReport.java
+++ b/src/main/java/org/codehaus/mojo/antlr/AntlrHtmlReport.java
@@ -26,13 +26,13 @@
 import java.util.Locale;
 
 import org.apache.maven.doxia.siterenderer.Renderer;
-import org.apache.maven.doxia.siterenderer.RendererException;
 import org.apache.maven.doxia.siterenderer.sink.SiteRendererSink;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.reporting.AbstractMavenReportRenderer;
 import org.apache.maven.reporting.MavenReport;
 import org.apache.maven.reporting.MavenReportException;
+import org.apache.maven.reporting.sink.SinkFactory;
 import org.apache.maven.wagon.PathUtils;
 import org.codehaus.doxia.sink.Sink;
 import org.codehaus.plexus.i18n.I18N;
@@ -218,20 +218,10 @@
     {
         try
         {
-            SiteRendererSink sink = siteRenderer.createSink( getReportOutputDirectory(), getOutputName() + ".html" );
+            SiteRendererSink sink = SinkFactory.createSink( getReportOutputDirectory(), getOutputName() + ".html" );
 
             generate( sink, Locale.getDefault() );
         }
-        catch ( RendererException e )
-        {
-            throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
-                + " report generation.", e );
-        }
-        catch ( IOException e )
-        {
-            throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
-                + " report generation.", e );
-        }
         catch ( MavenReportException e )
         {
             throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
--- a/src/main/java/org/codehaus/mojo/antlr/AbstractAntlrMojo.java
+++ b/src/main/java/org/codehaus/mojo/antlr/AbstractAntlrMojo.java
@@ -82,7 +82,7 @@
     /**
      * The maven project's helper.
      *
-     * @parameter expression="${component.org.apache.maven.project.MavenProjectHelper}"
+     * @component role="org.apache.maven.project.MavenProjectHelper"
      * @required
      * @readonly
      */
